Summary fails for larger input

I am running the summary on multiple  sarif files and it seems that it does not correctly handle larger input

```
Run sarif summary ./*_scan.sarif -o ./hdf/issues.txt
Traceback (most recent call last):
  File "/home/runner/.local/bin/sarif", line 8, in <module>
    sys.exit(main())
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/cmdline/main.py", line 61, in main
    exitcode = args.func(args)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/cmdline/main.py", line 401, in _summary_command
    summary_op.generate_summary(input_files, output, multiple_file_output)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/operations/summary_op.py", line 39, in generate_summary
    summary_lines = _generate_summary(input_files)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/operations/summary_op.py", line 64, in _generate_summary
    issue_type_histogram = report.get_issue_type_histogram_for_severity(severity)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/issues_report.py", line 138, in get_issue_type_histogram_for_severity
    self._group_records_by_key()
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/issues_report.py", line 45, in _group_records_by_key
    key = combine_record_code_and_description(record)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/sarif_file_utils.py", line 52, in combine_record_code_and_description
    return combine_code_and_description(record["Code"], record["Description"])
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/sarif_file_utils.py", line 33, in combine_code_and_description
    shorter_description = textwrap.shorten(
  File "/usr/lib/python3.10/textwrap.py", line 414, in shorten
    return w.fill(' '.join(text.strip().split()))
  File "/usr/lib/python3.10/textwrap.py", line 371, in fill
    return "\n".join(self.wrap(text))
  File "/usr/lib/python3.10/textwrap.py", line 362, in wrap
    return self._wrap_chunks(chunks)
  File "/usr/lib/python3.10/textwrap.py", line 263, in _wrap_chunks
    raise ValueError("placeholder too large for max width")
ValueError: placeholder too large for max width
Error: Process completed with exit code 1.
```

--- a/sarif/sarif_file_utils.py
+++ b/sarif/sarif_file_utils.py
@@ -23,7 +23,9 @@ def combine_code_and_description(code: str, description: str) -> str:
         code = code.strip()
         length_budget -= len(code) + 1
     # Allow extra space when truncating for continuation characters
-    length_budget_pre_continuation = length_budget - 20
+    length_budget_pre_continuation = length_budget - 5
+    if length_budget_pre_continuation < 10:
+        return code
     if description:
         if "\n" in description:
             description = description[: description.index("\n")]
@@ -33,7 +35,7 @@ def combine_code_and_description(code: str, description: str) -> str:
             shorter_description = textwrap.shorten(
                 description, width=length_budget_pre_continuation, placeholder=" ..."
             )
-            if len(shorter_description) < 40:
+            if len(shorter_description) < length_budget_pre_continuation - 40:
                 description = description[:length_budget_pre_continuation] + " ..."
             else:
                 description = shorter_description
@@ -41,7 +43,7 @@ def combine_code_and_description(code: str, description: str) -> str:
             return f"{code.strip()} {description}"
         return description
     if code:
-        return code.strip()
+        return code
     return "<NONE>"
 
 

--- a/sarif/sarif_file_utils.py
+++ b/sarif/sarif_file_utils.py
@@ -21,10 +21,12 @@ def combine_code_and_description(code: str, description: str) -> str:
     length_budget = 120
     if code:
         code = code.strip()
-        length_budget -= len(code) + 1
+        length_budget -= len(code) + 1  # Allow issue code and space character
+    continuation_placeholder = " ..."
     # Allow extra space when truncating for continuation characters
-    length_budget_pre_continuation = length_budget - 5
+    length_budget_pre_continuation = length_budget - len(continuation_placeholder)
     if length_budget_pre_continuation < 10:
+        # Don't include description if it would be very short due to long code
         return code
     if description:
         if "\n" in description:
@@ -33,10 +35,16 @@ def combine_code_and_description(code: str, description: str) -> str:
     if description:
         if len(description) > length_budget:
             shorter_description = textwrap.shorten(
-                description, width=length_budget_pre_continuation, placeholder=" ..."
+                description,
+                width=length_budget_pre_continuation,
+                placeholder=continuation_placeholder,
             )
             if len(shorter_description) < length_budget_pre_continuation - 40:
-                description = description[:length_budget_pre_continuation] + " ..."
+                # Word wrap shortens the description significantly, so truncate mid-word instead
+                description = (
+                    description[:length_budget_pre_continuation]
+                    + continuation_placeholder
+                )
             else:
                 description = shorter_description
         if code:

--- a/tests/test_sarif_file_utils.py
+++ b/tests/test_sarif_file_utils.py
@@ -1,12 +1,15 @@
 from sarif import sarif_file_utils
 
 
-def test_combine_code_and_description():
+def test_combine_code_and_description_short():
     cd = sarif_file_utils.combine_code_and_description(
         "ABC123", "Some short description"
     )
     assert cd == "ABC123 Some short description"
     assert len(cd) <= 120
+
+
+def test_combine_code_and_description_long_desc():
     cd = sarif_file_utils.combine_code_and_description(
         "ABC123", " ".join(f"blah{i}" for i in range(1, 30))
     )
@@ -15,10 +18,13 @@ def test_combine_code_and_description():
         == "ABC123 blah1 blah2 blah3 blah4 blah5 blah6 blah7 blah8 blah9 blah10 blah11 blah12 blah13 blah14 blah15 blah16 ..."
     )
     assert len(cd) <= 120
+
+
+def test_combine_code_and_description_long_code():
     long_code = "".join(f"A{i}" for i in range(1, 36)) + "BC"
     assert (
         len(long_code) == 98
-    ), "98 is right length to hit'placeholder too large for max width' without defensive code"
+    ), "98 is right length to hit 'placeholder too large for max width' without defensive code"
     cd = sarif_file_utils.combine_code_and_description(
         long_code, "wow that's a long code"
     )

--- a/tests/test_sarif_file_utils.py
+++ b/tests/test_sarif_file_utils.py
@@ -1,0 +1,31 @@
+from sarif import sarif_file_utils
+
+
+def test_combine_code_and_description():
+    cd = sarif_file_utils.combine_code_and_description(
+        "ABC123", "Some short description"
+    )
+    assert cd == "ABC123 Some short description"
+    assert len(cd) <= 120
+    cd = sarif_file_utils.combine_code_and_description(
+        "ABC123", " ".join(f"blah{i}" for i in range(1, 30))
+    )
+    assert (
+        cd
+        == "ABC123 blah1 blah2 blah3 blah4 blah5 blah6 blah7 blah8 blah9 blah10 blah11 blah12 blah13 blah14 blah15 blah16 ..."
+    )
+    assert len(cd) <= 120
+    long_code = "".join(f"A{i}" for i in range(1, 36)) + "BC"
+    assert (
+        len(long_code) == 98
+    ), "98 is right length to hit'placeholder too large for max width' without defensive code"
+    cd = sarif_file_utils.combine_code_and_description(
+        long_code, "wow that's a long code"
+    )
+    assert cd == f"{long_code} wow that's a ..."
+    assert len(cd) <= 120
+    long_code = "".join(f"A{i}" for i in range(1, 50))
+    cd = sarif_file_utils.combine_code_and_description(
+        long_code, "wow that's a long code"
+    )
+    assert cd == long_code

